model/Report/ReportComparator.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoSystemStatus\model\Report;
+
+use common_report_Report as Report;
+use oat\taoSystemStatus\model\Check\CheckInterface;
+
+/**
+ * Class ReportComparator
+ *
+ * Comparator is used to get difference between 2 reports and find degradations or restorations of system status
+ *
+ * @package oat\taoSystemStatus\model\Report
+ */
+class ReportComparator
+{
+    /**
+     * @var Report
+     */
+    private $oldReport;
+
+    /**
+     * @var Report
+     */
+    private $newReport;
+
+    const REPORTS_MAP = [
```

Comment:
After checking the CR in more depth, I've realized this map is about assigning weight to each report type. Hence I recommend a more expressive name, like `REPORT_WEIGHT_MAP`